sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -42,7 +43,7 @@ internal EventGridExtensionConfigProvider(Func<EventGridAttribute, IAsyncCollect
         // default constructor
         public EventGridExtensionConfigProvider(ILoggerFactory loggerFactory)
         {
-            _converter = (attr => new EventGridAsyncCollector(new EventGridPublisherClient(new Uri(attr.TopicEndpointUri), new EventGridSharedAccessSignatureCredential(attr.TopicKeySetting))));
+            _converter = (attr => new EventGridAsyncCollector(new EventGridPublisherClient(new Uri(attr.TopicEndpointUri), new AzureKeyCredential(attr.TopicKeySetting))));
```

Comment:
Unrelated to this PR: Is there something we could have done in the EventGrid SDK Documentation itself to make it clear which of these you needed to use?